hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/recordcache/RecordCacheImpl.java

```diff
@@ -324,14 +325,18 @@ public List<TransactionRecord> getRecords(@NonNull final AccountID accountID) {
 
         // While we still need to gather more records, collect them from the different histories.
         final var records = new ArrayList<TransactionRecord>(maxRemaining);
-        for (final var transactionID : transactionIDs) {
-            final var history = histories.get(transactionID);
-            if (history != null) {
-                final var recs = history.orderedRecords();
-                records.addAll(recs.size() > maxRemaining ? recs.subList(0, maxRemaining) : recs);
-                maxRemaining -= recs.size();
-                if (maxRemaining <= 0) break;
+        try {
```

Comment:
Or why not just make the local `records` be a `ConcurrentLinkedQueue` and pretty much be done with it.  (Except then L342 becomes `records.stream().sorted(...)`.)

Reply:
The CME arises from traversing the iterator constructed by this `for`-`each`,
```
            for (final var transactionID : transactionIDs) {
```
...which I'm not quite sure how to work around. But this query is literally unused afaik. 🤷